gpii/node_modules/flowManager/test/SolutionsRegistryDataSourceTests.js

```diff
@@ -0,0 +1,320 @@
+// fetched from the repository
+fluid.defaults("gpii.tests.solutionsRegistry.dataSource.local.mockRepository.env", {
+    gradeNames: ["fluid.test.testEnvironment"],
+    components: {
+        tester: {
+            type: "gpii.tests.solutionsRegistry.local.mockRepository.testCaseHolder"
+        }
+    }
+});
+kettle.test.bootstrap("gpii.tests.solutionsRegistry.dataSource.local.mockRepository.env");
+
+// Test component for the LFM solutions registry data source where it fails to
+// load from the source code repository.
+fluid.defaults("gpii.tests.solutionsRegistry.dataSource.local.failLoadFromRepository", {
+    gradeNames: ["gpii.flowManager.solutionsRegistry.dataSource.local"],
+    components: {
+        revisionRequester: {
+            type: "gpii.flowmanager.revisionRequester",
+            options: {
+                cloudURL: "http://gpii.net"
+            }
+        }
+    },
+    listeners: {
+        "loadSolutions.loadFromLocalDisk": {
+            listener: "{that}.loadFromLocalDisk",
+            priority: "first"
+        },
+        "loadSolutions.getRevision": {
+            listener: "{revisionRequester}.getRevision",
+            priority: "after:loadFromLocalDisk"
+        },
+        "loadSolutions.loadFromRepository": {
+            listener: "gpii.tests.solutionsRegistry.dataSource.local.nullRepoLoad",
+            args: ["{that}"],
+            priority: "after:getRevision"
+        },
+        "loadSolutions.solutionsLoaded": {
+            listener: "{that}.events.solutionsRegistryReady.fire",
+            priority: "after:loadFromRepository"
+        }
+    }
+});
+
+gpii.tests.solutionsRegistry.dataSource.local.nullRepoLoad = function (that) {
+    var promise = fluid.promise();
+    that.repositorySolutionsRegistry = null;
+    promise.resolve(that.repositorySolutionsRegistry);
+    return promise;
+};
+
+fluid.defaults("gpii.tests.solutionsRegistry.local.noRepository.testCaseHolder", {
+    gradeNames: "fluid.test.testCaseHolder",
+    components: {
+        solutionsRegistryNullRepoLoad: {
+            type: "gpii.tests.solutionsRegistry.dataSource.local.failLoadFromRepository",
+            options: {
+                path: gpii.tests.solutionsRegistry.path
+            }
+        }
+    },
+    modules: [{
+        name: "Solutions registry used by LFM - only file system based solutions available",
+        expect: 2,
+        tests: [{
+            name: "Solutions registry used with LFM - no solutions loaded from repository",
+            sequence: [
+                { funcName: "fluid.log", args: ["Solutions Registry No Repo Load, getting named 'repository' registry -- START"] },
+                {
+                    event: "{solutionsRegistryNullRepoLoad}.events.solutionsRegistryReady",
+                    listener: "jqUnit.assert",
+                    args: ["LFM solutionsRegistryNullRepoLoad ready"]
+                }, {
+                    funcName: "{solutionsRegistryNullRepoLoad}.get",
```

Comment:
Same as a previous comment too. One set of `get()` call and the verification should be enough, rather than 2 sets.

Reply:
Same reply as above.  And, in this case, since it mocks a failure to load solutions from the source code repository, this test is checking the fallback of using the ones on the local file system.  It's arguable that the second call to `get()` is redundant since it's testing the same thing as the second `get()` in the mock repository load test case, above.